doc/schema/libdoc.02.xsd

```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
+  <xs:element name="keywordspec" type="KeywordSpec" />
+  <xs:complexType name="KeywordSpec">
+    <xs:sequence>
+      <xs:element name="version" type="xs:string" />
+      <xs:element name="scope" type="LibraryScope" minOccurs="0" />
+      <xs:element name="namedargs" type="SupportNamedArgs" minOccurs="0" />
+      <xs:element name="doc" type="xs:string" />
+      <xs:element name="init" type="Init" minOccurs="0" />
+      <xs:element name="kw" type="Keyword" minOccurs="0" maxOccurs="unbounded" />
+    </xs:sequence>
+    <xs:attribute name="name" type="xs:string" use="required" />
+    <xs:attribute name="type" type="LibraryType" use="required" />
+    <xs:attribute name="format" type="DocumentationFormat" use="required" />
+    <xs:attribute name="scope" type="LibraryScope" use="required" />
+    <xs:attribute name="namedargs" type="xs:boolean" use="required" />
+    <xs:attribute name="generated" type="GenerationDate" use="required" />
+    <xs:attribute name="specversion" type="SpecVersion" use="required" />
+    <xs:attribute name="source" type="xs:string" use="optional" />
+    <xs:attribute name="lineno" type="xs:positiveInteger" use="optional" />
+  </xs:complexType>
+  <xs:complexType name="Init">
+    <xs:sequence>
+      <xs:element name="arguments" type="Arguments" />
+      <xs:element name="doc" type="xs:string" />
+      <xs:element name="tags" type="Tags" />
+    </xs:sequence>
+    <xs:attribute name="deprecated" type="xs:boolean" use="optional" />
+    <xs:attribute name="source" type="xs:string" use="optional" />
+    <xs:attribute name="lineno" type="xs:positiveInteger" use="optional" />
+  </xs:complexType>
+  <xs:complexType name="Keyword">
+    <xs:sequence>
+      <xs:element name="arguments" type="Arguments" />
+      <xs:element name="doc" type="xs:string" />
+      <xs:element name="tags" type="Tags" />
+    </xs:sequence>
+    <xs:attribute name="name" type="xs:string" use="required"/>
+    <xs:attribute name="deprecated" type="xs:boolean" use="optional" />
+    <xs:attribute name="source" type="xs:string" use="optional" />
+    <xs:attribute name="lineno" type="xs:positiveInteger" use="optional" />
```

Comment:
Here we could have a comment that "source" and "lineno" are omitted if getting them from the library has failed and "source" is also omitted if it is same as the library has. Especially the latter piece of information is important when creating tools that use this info for something.

Reply:
please put your comment there. This would definitely have a better wording as if i do it.